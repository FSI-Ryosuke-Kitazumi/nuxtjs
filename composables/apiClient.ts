export const apiClient = (url: string) => {
  const nuxtApp = useNuxtApp();
  
  return useFetch(url, {
    key: url,
    getCachedData: (key) => nuxtApp.payload.data[key] || nuxtApp.static.data[key],
  });
}
