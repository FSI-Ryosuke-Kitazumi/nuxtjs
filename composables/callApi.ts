import type { User } from "~/mocks/handlers";

export function useCallApi() {
  return useFetch<User>("https://example.com/user");
}
