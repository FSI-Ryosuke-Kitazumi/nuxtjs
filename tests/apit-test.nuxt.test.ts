import { mountSuspended } from "@nuxt/test-utils/runtime";
import { expect, test } from "vitest";
import { useCallApi } from "~/composables/callApi";
import ApiTest from "../pages/api-test.vue";

test("snapshot", async () => {
  const component = await mountSuspended(ApiTest);
  expect(component.html()).toMatchSnapshot();
});

test("should first", async () => {
  const { data } = await useCallApi();
  expect(data.value?.firstName).toBe("John");
});
