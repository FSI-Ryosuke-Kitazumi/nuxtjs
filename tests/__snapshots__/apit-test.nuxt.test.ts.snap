// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
"<h1>api-test</h1>
 John"
`;
